api/annotate/traces.go

```diff
@@ -0,0 +1,160 @@
+package annotate
+
+import (
+	"encoding/json"
+	"github.com/logzio/ezkonnect-server/api"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"net/http"
+)
+
+// TracesResourceRequest ResourceRequest is the JSON body of the POST request
+// It contains the name, kind, namespace, telemetry type and action of the resource
+// name: name of the resource
+// kind: kind of the resource (deployment or statefulset)
+// namespace: namespace of the resource
+// action: action to perform (add or delete)
+type TracesResourceRequest struct {
+	Name      string `json:"name"`
+	Kind      string `json:"controller_kind"`
+	Namespace string `json:"namespace"`
+	Action    string `json:"action"`
+}
+
+// TracesResourceResponse  is the JSON response of the POST request
+// It contains the name, kind, namespace and updated annotations of the resource
+// name: name of the resource
+// kind: kind of the resource (deployment or statefulset)
+// namespace: namespace of the resource
+// updated_annotations: updated annotations of the resource
+type TracesResourceResponse struct {
+	Name               string            `json:"name"`
+	Namespace          string            `json:"namespace"`
+	Kind               string            `json:"controller_kind"`
+	UpdatedAnnotations map[string]string `json:"updated_annotations"`
+}
+
+func UpdateTracesResourceAnnotations(w http.ResponseWriter, r *http.Request) {
+	logger := api.InitLogger()
+	// Decode JSON body
+	var resources []TracesResourceRequest
+	err := json.NewDecoder(r.Body).Decode(&resources)
+	if err != nil {
+		logger.Error("Error decoding JSON body", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	// Get the Kubernetes config
+	config, err := api.GetConfig()
+	if err != nil {
+		logger.Error("Error getting Kubernetes config", err)
+		http.Error(w, "Error getting Kubernetes config", http.StatusInternalServerError)
```

Comment:
For maintenance purposes - would it be better if we'd have the message in a variable, and refer to it in both logger & http error?